src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -118,6 +118,18 @@ internal static SafeFreeCredentials AcquireCredentialsHandle(string package, boo
                 ((inSecurityBufferArray != null && inSecurityBufferArray.Length != 0) ? inSecurityBufferArray[0] : null),
                 outSecurityBuffer,
                 ref contextFlags);
+
+            // Confidentiality flag should not be set if not requested
+            if(status == SecurityStatusPal.CompleteNeeded)
+            {
+                ContextFlagsPal mask = ContextFlagsPal.Confidentiality;
+                if((requestedContextFlags & mask) != (contextFlags & mask))
```

Comment:
Do the flags need to match exactly, or is any overlap sufficient? i.e. instead of:
```C++
if ((requestedContextFlags & mask) != (contextFlags & mask))
```
should it be:
```C++
if (((requestedContextFlags & contextFlags) & mask) != 0)
```
? (I don't know what semantics are desired, so just asking.)